taichi/transforms/redundancy_in_loop_elimination.cpp

```diff
@@ -0,0 +1,146 @@
+#include "taichi/ir/ir.h"
+#include "taichi/ir/statements.h"
+#include "taichi/ir/transforms.h"
+#include "taichi/ir/visitors.h"
+#include "taichi/system/profiler.h"
+
+#include <stack>
+
+TLANG_NAMESPACE_BEGIN
+
+//#define LOOP_INVARIANT_ONLY_ALLOWING_LOOPS
```

Comment:
I think this can be a `CompileConfig` -- something like `move_loop_invariant_outside_if`, with default value `False`. See compile_config.h for other examples.